### ISSUE TITLE
Ginormous tick marks on retina display

### ISSUE THREAD

**Issue by nathanielvirgo** [JuliaPlots/Makie.jl#362]
By typing this in a notebook, I can open a plot window. 
```
scene = Scene(resolution = (800, 800))
display(Makie.image!(scene, rand(200,200)))`
```

I'm using a Mac and it plots at full "retina" resolution, which is really nice. However, the fonts and tick marks are enormous:
![image](https://user-images.githubusercontent.com/6964676/58523544-b6800e80-81ff-11e9-8117-e85458140e6f.png)

I can reduce the font sizes with
```
scene[Axis][:ticks][:textsize] = (3,3)
scene[Axis][:names][:textsize] = (3,3)
```
but the tick marks themselves are still massive:
![image](https://user-images.githubusercontent.com/6964676/58523597-01018b00-8200-11e9-8df8-86bbc84ea684.png)

I guess this is because, for some reason, on the retina display it thinks the plot has a really small physical size. But there isn't an obvious way to change this (at least, I couldn't find it in the documentation or by poking around). Is there some way to make the sizes of the fonts and tickmarks appear generally more sensible?

**Comment by mkborregaard** [comment]
I can replicate (on my Mac with Retina screen), but only when specifying the resolution - if I don't specify the resolution the font looks fine. Not really sure what may cause this, could be interesting to have a report from someone not on a Mac.

Btw it's not the tick marks that are big, it's the grid lines extending out past the image area. Makie's default Theme does not show axes or tick marks. Maybe grid lines should be turned off for heatmap/image plots?

**Comment by AndiMD** [comment]
I can confirm on Kubuntu 18.04, with a regular 1920x1080 display. Specifying the resolution does not help in my case.
I would expect gridlines by default, IIRC this is what matplotlib and matlab do.

**Comment by asinghvi17** [comment]
@AndiMD, what do you mean by `gridlines` - the extensions out of the plot limits or something else?

**Comment by AndiMD** [comment]
@asinghvi17 Yes, I guess "tickmarks" would have been the appropriate term. For example, the default matplotlib.pyplot.imshow yields this:
![Screenshot_20190620_143229](https://user-images.githubusercontent.com/24719218/59849563-411fdd80-9368-11e9-8aa2-c3a0932a5783.png)

**Comment by asinghvi17** [comment]
Ah, I see.  That's a consequence of how we draw the axis lines.  Once the recipe and layout system are fully in place, we will probably get to that...

**Comment by aramirezreyes** [comment]
Is there any update on this?

**Comment by SimonWoods** [comment]
I have the same issue on Windows, but worse. The grid extends by a full grid square on every side of the image.
![Clipboard01](https://user-images.githubusercontent.com/4841090/75073931-70d44600-54f2-11ea-9a65-b3a411fb1a7b.png)

**Comment by aramirezreyes** [comment]
@SimonWoods I think there is currently no remedy for this with only Makie, but you can try working with MakieLayout. Take a look at https://github.com/jkrumbiegel/MakieLayout.jl . Hopefully both packages will be integrated soon!

**Comment by asinghvi17** [comment]
Yep, they'll be integrated at Vizcon!  The extra-long tickmarks are sort of baked into the axis drawing code, so any attempt to remove them will be extra annoying.  IMO, it's not worth the effort considering that MakieLayout will become part of Makie so soon...

**Comment by SimonWoods** [comment]
> @SimonWoods I think there is currently no remedy for this with only Makie, but you can try working with MakieLayout. Take a look at https://github.com/jkrumbiegel/MakieLayout.jl . Hopefully both packages will be integrated soon!

Yeah that works brilliantly, thank you!
```julia
using MakieLayout
scene, layout = layoutscene(1, 1, 30, resolution=(500,500))
la = LAxis(scene)
tightlimits!(la)
la.aspect = DataAspect()
layout[1,1] = la
image!(la, randn(200,200))
scene
```
![Clipboard01](https://user-images.githubusercontent.com/4841090/75078101-9d8d5b00-54fc-11ea-9eb7-0ba90a099cb0.png)
